Extension/src/LanguageServer/extension.ts

```diff
@@ -82,6 +99,113 @@ export function activate(activationEventOccurred: boolean): void {
+        const languageAssociationFilter: (info: configs.CompilerInfo) => boolean = (info: configs.CompilerInfo): boolean => {
+            return (info.languageAssociation === 'cpp' && activeFileIsCpp) || (info.languageAssociation === 'c' && activeFileIsC);
+        };
+        compilerPaths = compilerInfo.filter(languageAssociationFilter).map<string>(info => { return info.path; });
+
+        if (userCompilerPath) {
+            compilerPaths.push(userCompilerPath);
+        }
+
+        let map: Map<string, string> = new Map<string, string>();
+        compilerPaths.forEach(value => {
+            const basename: string = path.basename(value);
+            map.has(basename) ? map[basename] = value : map.set(basename, value);
+        });
+        compilerPaths = [...map.values()];
+    } else if (userCompilerPath) {
+        compilerPaths = [userCompilerPath];
+    }
+
+    if (!compilerPaths) {
+        telemetry.logLanguageServerEvent('noCompilerFound'); // Don't prompt a message yet until we can make a data-based decision
+        // Display a message prompting the user to install compilers if none were found
+        // const dontShowAgain: string = "Don't Show Again";
+        // const learnMore: string = "Learn More";
+        // const message: string = "No C/C++ compiler found on the system. Please install a C/C++ compiler to use the C/Cpp: build active file tasks.";
+
+        // let showNoCompilerFoundMessage: PersistentState<boolean> = new PersistentState<boolean>("CPP.showNoCompilerFoundMessage", true);
+        // if (showNoCompilerFoundMessage) {
+        //     vscode.window.showInformationMessage(message, learnMore, dontShowAgain).then(selection => {
+        //         switch (selection) {
+        //             case learnMore:
+        //                 const uri: vscode.Uri = vscode.Uri.parse(`https://go.microsoft.com/fwlink/?linkid=864631`);
+        //                 vscode.commands.executeCommand('vscode.open', uri);
+        //                 break;
+        //             case dontShowAgain:
+        //                 showNoCompilerFoundMessage.Value = false;
+        //                 break;
+        //             default:
+        //                 break;
+        //         }
+        //     });
+        // }
+        return [];
+    }
+
+    // The build task output file should include a '.exe' extension on Windows
+    let platformInfo: PlatformInformation = await PlatformInformation.GetPlatformInformation();
+    let exeName: string;
+    if (platformInfo.platform === 'win32') {
+        exeName = '${fileBasenameNoExtension}.exe';
+    } else {
+        exeName = '${fileBasenameNoExtension}';
+    }
+
+    // Generate tasks
+    let result: vscode.Task[] = [];
+    compilerPaths.forEach(compilerPath => {
+        const taskName: string = path.basename(compilerPath) + " build active file";
+        const args: string[] = ['-g', '${fileDirname}/${fileBasename}', '-o', '${fileDirname}/' + exeName];
```

Comment:
To avoid accidental collisions, should the output exes be put in a \bin folder inside the directory?

Reply:
Not a bad idea. Another thought is putting it into the `.vscode` folder. @bobbrow?

Reply:
I would be against `.vscode` as that seems to be where configuration and extension data is going and not anything to do with the customer's coding experience. For me it would be between `/bin` or `/out`.